ui/apps/platform/src/Containers/Vulnerabilities/NodeCves/Node/CVEsTable.tsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { gql } from '@apollo/client';
+import { pluralize } from '@patternfly/react-core';
+import { Table, Thead, Tr, Th, Td, ExpandableRowContent, Tbody } from '@patternfly/react-table';
+
+import CvssFormatted from 'Components/CvssFormatted';
+import TbodyUnified from 'Components/TableStateTemplates/TbodyUnified';
+import VulnerabilityFixableIconText from 'Components/PatternFly/IconText/VulnerabilityFixableIconText';
+import { TableUIState } from 'utils/getTableUIState';
+import useSet from 'hooks/useSet';
+
+import VulnerabilitySeverityIconText from 'Components/PatternFly/IconText/VulnerabilitySeverityIconText';
+import {
+    getIsSomeVulnerabilityFixable,
+    getHighestVulnerabilitySeverity,
+} from '../../utils/vulnerabilityUtils';
+import { getNodeEntityPagePath } from '../../utils/searchUtils';
+import NodeComponentsTable, {
+    NodeComponent,
+    nodeComponentFragment,
+} from '../components/NodeComponentsTable';
+
+export const nodeVulnerabilityFragment = gql`
+    ${nodeComponentFragment}
+    fragment NodeVulnerabilityFragment on NodeVulnerability {
+        cve
+        summary
+        cvss
+        scoreVersion
+        nodeComponents(query: $query) {
+            ...NodeComponentFragment
+            nodeVulnerabilities(query: $query) {
+                severity
+                isFixable
+                fixedByVersion
+            }
+        }
+    }
+`;
+
+export type NodeVulnerability = {
+    cve: string;
+    summary: string;
+    cvss: number;
+    scoreVersion: string;
+    nodeComponents: (NodeComponent & {
```

Comment:
Would exporting this as another type from the `NodeComponentsTable` be worthwhile? That way, it would be easier if you needed to reuse this type.

Reply:
Ah got it. I left that as a separate type since the `NodeComponentsTable` doesn't use any of those fields, they are strictly used in the `CVEsTable` component. I've been trying to keep the query fragment and resulting type as close to the component that needs them, and then aggregate these values at a higher level. This way the component is the element of reuse instead of the type/fragment, and it reduces the need for larger refactoring if something changes. e.g. If we need to add a new field that the child component depends on, we can do it in that component directly (both fragment and type), and whatever tree of parents are above it will not require a change since they import the data dependency from the child.